pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -278,12 +282,29 @@ private static boolean isAuthException(ConnectionException e) {
      * @throws DDPException if error getting a raw connection
      */
     public static <R, X extends Exception> R withTxn(DB db, HandleCallback<R, X> callback) throws X {
+        // hopefully temporary code to detect long-running connections
+        long startTime = System.currentTimeMillis();
         try {
             try (Handle h = openJdbiWithAuthRetry(db)) {
-                return h.inTransaction(callback);
+                R res =  h.inTransaction(callback);
+                long endTime = System.currentTimeMillis();
+                // 30s threshold
+                if (endTime - startTime > 30000) {
+                    logger.warn("DB transaction open for > 30s\n {}",
+                            stackTraceToString(Thread.currentThread().getStackTrace()));
```

Comment:
Are you familiar with [StackWalker](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StackWalker.html)?  Nice addition and it purports to speed performance of this use case.